pkg/operator/discover/discover.go

```diff
@@ -72,15 +70,15 @@ func New(clientset kubernetes.Interface) *Discover {
 }
 
 // Start the discover
-func (d *Discover) Start(ctx context.Context, namespace, discoverImage, securityAccount string, useCephVolume bool) error {
-	err := d.createDiscoverDaemonSet(ctx, namespace, discoverImage, securityAccount, useCephVolume)
+func (d *Discover) Start(ctx context.Context, namespace, discoverImage, securityAccount string, data map[string]string, useCephVolume bool) error {
+	err := d.createDiscoverDaemonSet(ctx, namespace, discoverImage, securityAccount, data, useCephVolume)
 	if err != nil {
 		return fmt.Errorf("failed to start discover daemonset. %v", err)
 	}
 	return nil
 }
 
-func (d *Discover) createDiscoverDaemonSet(ctx context.Context, namespace, discoverImage, securityAccount string, useCephVolume bool) error {
+func (d *Discover) createDiscoverDaemonSet(ctx context.Context, namespace, discoverImage, securityAccount string, data map[string]string, useCephVolume bool) error {
```

Comment:
For backward compatibility the env vars must also work. IIRC this map would also contain env vars, but please just test that the env vars still work.

Reply:
I don't see env setting in the configmap data
```yaml
get cm rook-ceph-operator-config  -oyaml
apiVersion: v1
data:
  CSI_CEPHFS_ATTACH_REQUIRED: "true"
  CSI_CEPHFS_FSGROUPPOLICY: File
  CSI_ENABLE_CEPHFS_SNAPSHOTTER: "true"
  CSI_ENABLE_CSIADDONS: "false"
  CSI_ENABLE_ENCRYPTION: "false"
  CSI_ENABLE_LIVENESS: "false"
  CSI_ENABLE_NFS_SNAPSHOTTER: "true"
  CSI_ENABLE_RBD_SNAPSHOTTER: "true"
  CSI_ENABLE_READ_AFFINITY: "false"
  CSI_ENABLE_TOPOLOGY: "false"
  CSI_FORCE_CEPHFS_KERNEL_CLIENT: "true"
  CSI_GRPC_TIMEOUT_SECONDS: "150"
  CSI_NFS_ATTACH_REQUIRED: "true"
  CSI_NFS_FSGROUPPOLICY: File
  CSI_PLUGIN_ENABLE_SELINUX_HOST_MOUNT: "false"
  CSI_PLUGIN_PRIORITY_CLASSNAME: system-node-critical
  CSI_PROVISIONER_PRIORITY_CLASSNAME: system-cluster-critical
  CSI_PROVISIONER_REPLICAS: "2"
  CSI_RBD_ATTACH_REQUIRED: "true"
  CSI_RBD_FSGROUPPOLICY: File
  ROOK_CEPH_ALLOW_LOOP_DEVICES: "false"
  ROOK_CEPH_COMMANDS_TIMEOUT_SECONDS: "15"
  ROOK_CSI_ALLOW_UNSUPPORTED_VERSION: "false"
  ROOK_CSI_ENABLE_CEPHFS: "true"
  ROOK_CSI_ENABLE_GRPC_METRICS: "false"
  ROOK_CSI_ENABLE_NFS: "false"
  ROOK_CSI_ENABLE_RBD: "true"
  ROOK_DISABLE_ADMISSION_CONTROLLER: "true"
  ROOK_ENABLE_DISCOVERY_DAEMON: "false"
  ROOK_LOG_LEVEL: INFO
  ROOK_OBC_WATCH_OPERATOR_NAMESPACE: "true"
  ROOK_WATCH_FOR_NODE_FAILURE: "true"
kind: ConfigMap
```

Reply:
I see that you're calling [k8sutil.GetValue()](https://github.com/rook/rook/blob/71f078c0fcfefabbe4a319226d0d65002a56c90f/pkg/operator/k8sutil/configmap.go#L64), so that is sufficient. That method will first look in the configmap settings, and if it's not found, then fall back to query for an environment variable on the operator.